### PR TITLE
Redesign deploy new agent page

### LINKED ISSUE #4205: Redesign of add agent page
## Summary

Update the way the "add agent wizard" works. Currently, we have:

cloud

## Requirements

We need to:
* separate the selection of each operating system, as the installation command might be different for each one. Even if the package is the same, we might separate OS versions if the installation method is different.
* select the correct init code for each operating system, helping the user in the selection of systemd / init
* use the correct install command for each agent

## Design
We need to propose a design for the selector if we reach the conclusion the current approach is too hard to use when a lot of OSes are shown, for example, we might want to select the OS and then the version in two steps.

## To do 

- [x] #4178 
- [x] #4204
- [x] #4109
- [x] #4210
- [x] #4222
- [x] #4318
- [x] #4484
- [x] #4573
- [x] #4850 
- [ ] #4718 
- [x] #4771
- [x] #4786
- [x] #4848 
- [x] #5466
- [x] https://github.com/wazuh/wazuh-splunk/issues/1406
- [ ] #5493 
- [x] #5531 

## New design
- [x] #5441 
- [x] #5455
- [x] #5547
- [x] #5435 
- [x] #5456
- [x] #5477 
- [x] #5489 
- [x] #5518
- [x] #5594
- [x] #5610
- [x] #5632

### PR DESCRIPTION
**Pull request by chantal-kelm**
### Description

This PR contains all the integrated PR's from the redesign of the deploy a new agent page.
 
### Issues Resolved

#4205 

### Evidence

Captura de pantalla 2023-06-30 a la(s) 14 37 04
Captura de pantalla 2023-06-30 a la(s) 14 37 13
Captura de pantalla 2023-06-30 a la(s) 14 37 17
Captura de pantalla 2023-06-30 a la(s) 14 34 35
Captura de pantalla 2023-06-30 a la(s) 14 35 29
Captura de pantalla 2023-06-30 a la(s) 14 35 35

### Test

Verify that everything is displayed as in the evidence image

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by jbiset — DISMISSED** [review]
LGTM
CR: :heavy_check_mark: 
Test: :heavy_check_mark: 

## Evidence

### Light theme
<details>

Chrome_lightTheme_1

Chrome_lightTheme_2

Chrome_lightTheme_3_with_wrong_server_address

Chrome_lightTheme_4

</details>

### Dark theme
<details>

Chrome_darkTheme_1

Chrome_darkTheme_2

Chrome_darkTheme_3_with_wrong_server_address

Chrome_darkTheme_4


</details>

**Comment by Machi3mfl** [comment]
## Status

This PR must be approved by @gdiazlo before merging it

**Comment by Machi3mfl** [comment]
## Latest requested changes

-  [ ]  Change the callout information text from `check our step` to `check our documentation` 

image

- [ ] Verify FDQN valid required labels quantity

image

- [ ] Add documentation links on information callouts (agent name, server address, etc)

image

- [ ] Adjust SO labels in SO cards

image

image

**Comment by Machi3mfl** [comment]
## Test Cases

1. Check the callout information text changes from check our step to check our documentation
2. Check the documentation link added on agent name yellow callout message
3. Verify FDQN valid required labels quantity (changed for 2 required labels)
4. Check the labels typography changed (Windows, Apple silicon)
5. Check the install command format for all the DEB and RPM distros (ignoring architecture)

**Test cases evidence in [comment](https://github.com/wazuh/wazuh-kibana-app/pull/5457#issuecomment-1658741129)**

File: plugins/main/public/controllers/register-agent/components/command-output/command-output.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  EuiSwitchEvent,
+  EuiText,
+} from '@elastic/eui';
+import React, { Fragment, useEffect, useState } from 'react';
+
+interface ICommandSectionProps {
+  commandText: string;
+  showCommand: boolean;
+  onCopy: () => void;
+  os?: 'WINDOWS' | string;
```
[the remark this answers is not in the document]
Comment (Machi3mfl) [reply plugins/main/public/controllers/register-agent/components/command-output/command-output.tsx]: Added in https://github.com/wazuh/wazuh-kibana-app/pull/5457/commits/cd5739c37a6d2f79b74d6184883b2c6095c9fb5a

**Review by jbiset — APPROVED** [review]
CR: :heavy_check_mark: 

## UI

### Test cases

1. :green_circle:  Check the callout information text changes from check our step to check our documentation

test1


2. :green_circle:  Check the documentation link added on agent name yellow callout message

test2

3. :green_circle: Verify FDQN valid required labels quantity (changed for 2 required labels)

test3_A

test3_B

test3_C

test3_D

test3_E

test3_F

4. :green_circle:  Check the labels typography changed (Windows, Apple silicon)

test4

5. :green_circle:  Check the install command format for all the DEB and RPM distros (ignoring architecture)

### RPM dist

```bash

curl -o wazuh-agent-4.4.5-1.x86_64.rpm https://packages.wazuh.com/4.x/yum/wazuh-agent-4.4.5-1.x86_64.rpm && sudo WAZUH_MANAGER='172.30.30.20' rpm -ihv wazuh-agent-4.4.5-1.x86_64.rpm

```

test5_B

### DEB dist

```bash

wget https://packages.wazuh.com/4.x/apt/pool/main/w/wazuh-agent/wazuh-agent_4.4.5-1_amd64.deb && sudo WAZUH_MANAGER='172.30.30.20' dpkg -i ./wazuh-agent_4.4.5-1_amd64.deb

```
test5_A

**Comment by Machi3mfl** [comment]
Backports are not necessary for the `4.6.0 release`.

**Pull request merged**